tests/unit/matrix/test_subtotals.py

```diff
@@ -580,98 +588,75 @@ def it_can_compute_a_subtotal_intersection_with_subtrahends(
 
         np.testing.assert_almost_equal(
             subtotals._subtotal_column(subtotal_),
             np.array([-1.2667117, 0.0, 0.7368146]),
         )
 
     def it_can_compute_a_subtotal_column_with_subtrahends(
-        self,
-        request,
-        subtotal_,
+        self, request, subtotal_, _base_weighted_counts_
     ):
         subtotal_.subtrahend_idxs = np.array([0])
         property_mock(request, ZscoreSubtotals, "_nrows", return_value=3)
-        subtotals = ZscoreSubtotals(None, None, None)
+        subtotals = ZscoreSubtotals(_base_weighted_counts_, None)
 
         np.testing.assert_almost_equal(
             subtotals._subtotal_column(subtotal_),
             np.array([np.nan, np.nan, np.nan]),
         )
 
-    def it_can_compute_a_subtotal_row_to_help(
-        self,
-        subtotal_,
-        _base_counts_prop_,
-        _table_margin_prop_,
-        cube_result_matrix_,
-    ):
+    def it_can_compute_a_subtotal_row_to_help(self, subtotal_, _base_weighted_counts_):
         subtotal_.addend_idxs = np.array([0, 1])
         subtotal_.subtrahend_idxs = np.array([])
-        _base_counts_prop_.return_value = np.arange(12).reshape(3, 4)
-        cube_result_matrix_.columns_margin = np.array([12, 15, 18, 21])
-        _table_margin_prop_.return_value = 66
-        subtotals = ZscoreSubtotals(None, None, cube_result_matrix_)
+        _base_weighted_counts_.weighted_counts = np.arange(12).reshape(3, 4)
+        _base_weighted_counts_.columns_margin = np.array([12, 15, 18, 21])
+        _base_weighted_counts_.table_margin = 66
+        subtotals = ZscoreSubtotals(_base_weighted_counts_, None)
 
         np.testing.assert_almost_equal(
             subtotals._subtotal_row(subtotal_),
             np.array([-0.7044435, -0.2161134, 0.2033553, 0.5833346]),
         )
 
     def it_can_compute_a_subtotal_row_with_subtrahends(
-        self,
-        request,
-        subtotal_,
+        self, request, subtotal_, _base_weighted_counts_
     ):
         subtotal_.subtrahend_idxs = np.array([0])
         property_mock(request, ZscoreSubtotals, "_ncols", return_value=4)
-        subtotals = ZscoreSubtotals(None, None, None)
+        subtotals = ZscoreSubtotals(_base_weighted_counts_, None)
 
         np.testing.assert_almost_equal(
             subtotals._subtotal_row(subtotal_),
             np.array([np.nan, np.nan, np.nan, np.nan]),
         )
 
-    def it_provides_access_to_the_table_margin_to_help(self, cube_result_matrix_):
-        cube_result_matrix_.table_margin = 42
-        subtotals = ZscoreSubtotals(None, None, cube_result_matrix_)
-        assert subtotals._table_margin == 42
-
     # --- fixture components -----------------------------------------
 
     @pytest.fixture
-    def _base_counts_prop_(self, request):
-        return property_mock(request, ZscoreSubtotals, "_base_counts")
+    def _base_weighted_counts_(self, request):
+        return instance_mock(request, _BaseWeightedCubeCounts)
```

Comment:
just `weighted_counts_` here I think. `_BaseWeightedCubeCounts` is the right class to use, but the name need not reflect that because in operation this value can (and will) be any of its subclasses.

Btw, an instance mock will in general never start with an underscore. That's just for method and property mocks on helper methods (that themselves start with an underscore).